zk/syncer/l1_syncer.go

```diff
@@ -267,8 +269,13 @@ func (s *L1Syncer) L1QueryHeaders(logs []ethTypes.Log) (map[uint64]*ethTypes.Hea
 	wg.Wait()
 
 	headersMap := map[uint64]*ethTypes.Header{}
-	for i := 0; i < len(headers); i++ {
-		headersMap[headers[i].Number.Uint64()] = headers[i]
+	for loop := true; loop; {
+		select {
+		case header := <-headersQueue:
+			headersMap[header.Number.Uint64()] = header
+		default:
+			loop = false
+		}
```

Comment:
selects don't guarantee ordering, could we miss some data here by quitting the loop early?

Reply:
instead, because we know how many headers we're expecting you can simply close the channel inside the processing loop when we know we've reached our count.  if we use an atomic.uint or something to keep count we won't miss anything.  Then in the select you could just have something like:
```
case header, ok := <- headersQueue:
if !ok {
break
}
```

Reply:
We cannot use !ok, because the headersQueue will never be closed until the function returns.

Also we know that headersQueue is full by the time we start reading from it, so we don't expect any more new elements to be added.

Reply:
changes are ready